src/uucore/src/lib/features/tokenize/sub.rs

```diff
@@ -261,7 +261,6 @@ impl SubParser {
                 }
                 x if legal_fields.binary_search(&x).is_ok() => {
                     self.field_char = Some(ch);
-                    self.text_so_far.push(ch);
```

Comment:
This unnecessary push cause wrong error message: `printf: %7bb invalid conversion specification\n`, which has one more `b`.